ql/src/java/org/apache/hadoop/hive/ql/io/parquet/serde/ArrayWritableObjectInspector.java

```diff
@@ -139,8 +160,9 @@ public Object getStructFieldData(final Object data, final StructField fieldRef)
     if (data instanceof ArrayWritable) {
       final ArrayWritable arr = (ArrayWritable) data;
       final StructFieldImpl structField = (StructFieldImpl) fieldRef;
-      if (structField.getIndex() < arr.get().length) {
-        return arr.get()[structField.getIndex()];
+      int index = isRoot ? structField.getIndex() : structField.adjustedIndex;
```

Comment:
Yes, when there's no pruning then we should still use the original index. This is the only place `StructFieldImpl` is initialized so I don't think the index is changed in any other places.